ayaneo-platform.c

```diff
@@ -388,24 +378,38 @@ static void ayaneo_led_mc_color(u8 *color)
 }
 
 /* Legacy methods */
-static void ayaneo_led_mc_set(u8 pos, u8 brightness)
+static void ayaneo_led_mc_set(u8 group, u8 pos, u8 brightness)
 {
-        write_to_ec(AYANEO_LED_MODE_REG, AYANEO_LED_MODE_WRITE);
-        write_to_ec(AYANEO_LED_POS, pos);
-        write_to_ec(AYANEO_LED_BRIGHTNESS, brightness);
-        mdelay(1);
-        write_to_ec(AYANEO_LED_MODE_REG, AYANEO_LED_MODE_WRITE_END);
+        if (!lock_global_acpi_lock())
+                return;
+
+        ec_write(AYANEO_LED_PWM_CONTROL, group);
+        ec_write(AYANEO_LED_POS, pos);
+        ec_write(AYANEO_LED_BRIGHTNESS, brightness);
+        ec_write(AYANEO_LED_MODE_REG, AYANEO_LED_MODE_WRITE);
+
+        if (!unlock_global_acpi_lock())
+                return;
+
+        mdelay(5);
```

Comment:
In my testing, mdelay(2); works fine here while nearly doubling the response time. This is a critical measurement for updating the RGB through steam in some games that I would like to keep as low as possible.

Reply:
Cool, I'm happy with that.
In the interests of perf, maybe we need to do these writes async and return early?